### ISSUE TITLE
No labels on custom video inference

### ISSUE THREAD

**Issue by Leoagnusdei** [facebookresearch/detectron2#3925]
Hi @ppwwyyxx and other contributors.
I trained my model on custom dataset of images. I created my dataset (files.jpeg and files.json) with Labelme.

## Instructions To Reproduce the Issue and Full Logs

```
!pip install pyyaml==5.1
import torch
!pip install detectron2 -f https://dl.fbaipublicfiles.com/detectron2/wheels/$CUDA_VERSION/torch$TORCH_VERSION/index.html

import detectron2
from detectron2.utils.logger import setup_logger
setup_logger()

import numpy as np
import os, json, cv2, random
from google.colab.patches import cv2_imshow
from detectron2 import model_zoo
from detectron2.engine import DefaultPredictor
from detectron2.config import get_cfg
from detectron2.utils.visualizer import Visualizer
from detectron2.data import MetadataCatalog, DatasetCatalog

!pip install labelme2coco
import labelme2coco

from detectron2.data.datasets import register_coco_instances
register_coco_instances("my_dataset_t", {},"/content/drive/MyDrive/........./annotation_dataaugmentation.json", "")

from detectron2.engine import DefaultTrainer

cfg = get_cfg()
cfg.merge_from_file(model_zoo.get_config_file("COCO-InstanceSegmentation/mask_rcnn_R_50_FPN_3x.yaml"))
cfg.DATASETS.TRAIN = ("my_dataset_t",)
cfg.DATASETS.TEST = ()
cfg.DATALOADER.NUM_WORKERS = 2
cfg.MODEL.WEIGHTS = model_zoo.get_checkpoint_url("COCO-InstanceSegmentation/mask_rcnn_R_50_FPN_3x.yaml")
cfg.SOLVER.IMS_PER_BATCH = 2
cfg.SOLVER.BASE_LR = 0.0038 
cfg.SOLVER.MAX_ITER = 300   
cfg.MODEL.ROI_HEADS.BATCH_SIZE_PER_IMAGE = 512   
cfg.MODEL.ROI_HEADS.NUM_CLASSES = 2  

os.makedirs(cfg.OUTPUT_DIR, exist_ok=True)
trainer = DefaultTrainer(cfg) 
trainer.resume_or_load(resume=False)
trainer.train()


!git clone https://github.com/facebookresearch/detectron2
%run detectron2/demo/demo.py --config-file /content/config.yml --video-input /content/drive/........ --confidence-threshold 0.1 --output test.mkv \
  --opts MODEL.WEIGHTS /content/output/model_final.pth
```

## Expected behavior:
After the training, I get a good prediction on a personal video, but I have just binary labels (0 for first istance and 1 for the second istance). However, in Labelme and in the annotation file, I have text labels. In fact, on image inference I  don't have this problem

How can I resolve for video inference?
Thanks

**Comment by ppwwyyxx** [comment]
There is no labels because `demo.py` the dataset is not registered in demo.py. This is expected, therefore closing.

See also https://github.com/facebookresearch/detectron2/issues/2279

**Comment by 413ph** [comment]
@Leoagnusdei Hi mate, I seems like I am also facing the exact same problem. Have you found any workaround on this problem?
